docs/src/main/asciidoc/validation.adoc

```diff
@@ -411,6 +411,44 @@ The scope you choose for your `ConstraintValidator` bean is important:
 - If the `ConstraintValidator` bean implements the `initialize(A constraintAnnotation)` method and depends on the state of the constraint annotation, use the `@Dependent` scope to make sure each annotation context has a separate and properly configured instance of the `ConstraintValidator` bean.
 ====
 
+When injecting beans relying on the runtime configuration, use `@Inject Instance<..>`. Since constraints are initialised
+at build time, it is not possible to completely preconfigure the constraint in the `initialize(..)` method,
+as the runtime information will be missing at that point. The `initialize(..)` method in such scenarios can still be used
+to read the constraint annotation parameters, and perform any work not relying on the runtime configuration.
+Hence, it is recommended that any heavy configuration work happens as part of the injected bean initialisation
+and methods used in the `ConstraintValidator#isValid(..)` implementation are as fast as possible:
+
+[source,java]
+----
+@ApplicationScoped
+public class MyConstraintValidator implements ConstraintValidator<MyConstraint, String> {
+
+    @Inject
+    Instance<MyService> service;
+
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        if (value == null) {
+            return true;
+        }
+
+        return service.get().validate(value);
+    }
+}
+
+@ApplicationScoped
+public class MyService {
+
+	public MyService(..) {
+		// perform all possible "initialisation" work
```

Comment:
You're talking about runtime configuration but then we don't see it in the example?

Reply:
hey 😃 👋🏻 
thanks for taking a look and for the suggestions!
Yeah ... I didn't want to make this example long and with some code that is, most likely, irrelevant to the user, so I thought I could get away with just a comment 🙈 😃.

would the following work here:
```java
@ApplicationScoped
public class MyService {

	private final Predicate<String> validationFunction;

	@Inject
	public MyService(MyRuntimeConfigProperties config) {
		// perform all possible "initialization" work, e.g.:
		if (config.complexValidationEnabled()) {
			validationFunction = s -> ...
		} else {
			validationFunction = String::isBlank;
		}
	}

	public boolean validate(String value) {
		// perform the validation
		return validationFunction.test(value);
	}
}
```

Reply:
Rename `MyRuntimeConfigProperties` to `MyRuntimeConfig` and we have a winner :).